docs/arm64_installation.md

```diff
@@ -0,0 +1,202 @@
+```
+
+Confirm the correct version of homebrew will be used for commands below.  The key here is looking
+for `/usr/local/bin` in the path where the `arm64` version is usually located in `/opt/homebrew/`
+```shell
+whereis brew
+# brew: /usr/local/bin/brew /usr/local/share/man/man1/brew.1
+```
+
+### 3- Brew install `x86` versions of pyenv and other dependencies
+
+```shell
+brew install pyenv xz
+```
+
+Again, for confirmation things are getting installed in the right place, you can use `whereis` to
+ensure it's installed to `/usr/local/bin`
+
+```shell
+whereis pyenv
+# pyenv: /usr/local/bin/pyenv /usr/local/share/man/man1/pyenv.1
+```
+
+### 4- Install pyenv plugin to support suffixes for install python versions
+
+This clones a repository and adds it to the `x86` pyenv installation.  This will allow in a following
+step to add a suffix to the python version to indicate it's an `x86` version
+
+```shell
+git clone https://github.com/AdrianDAlessandro/pyenv-suffix.git $(pyenv root)/plugins/pyenv-suffix
+```
+
+### 5- Install `x86` version of Python
+
+NOTE: when performing this step, it may warn that a python version of `3.11.8` already exists, which is okay.  This warning does not take into account that a suffix will be added to the python version in pyenv
+
+```shell
+PYENV_VERSION_SUFFIX="_x86" pyenv install 3.11.8
+```
+
+Confirm this new python version installed
+```shell
+pyenv versions
+#  system
+#  3.7.17
+#  3.10.12
+#  3.11.1
+#  3.11.4
+#  3.11.8
+#  3.11.8_x86   <---------- looking for this _x86 suffix
+#  3.12.2
+#  3.12.2_x86
+```
+
+### 6- Create `x86` python virtual environment
+
+Finally, with these steps complete, we can create a python virtual environment that will be 
+used like normal for all future development work.
+
+This is done instead of `make install` to ensure the correct `x86` python version is used.
+
+First, set an environment variable that will inform pyenv what version to use
+```shell
+export PYENV_VERSION=3.11.8_x86
+```
+
+Then, create the virtual environment
+```shell
+pipenv install --dev --python $(pyenv which python)
+```
+
+_Now_ you can run `make install` to ensure any additional work this performs is run (e.g. installing
+`pre-commit`)
+```shell
+make install
+```
+
+### 7- Install Oracle client
+
+Still using the `x86` version of brew, install the Oracle client
+
+```shell
+brew tap InstantClientTap/instantclient
+brew install instantclient-basiclite
+```
+
+Then, for the pipenv environment to use, set a required environment variable in the `.env`
+file.  This gives the python library enough context to know where to look for the Oracle client
+library files that were installed
+```shell
+DYLD_LIBRARY_PATH=/usr/local/lib:$DYLD_LIBRARY_PATH
+```
+
+### 8- Confirm virtual environment works as expected
+
+With the `x86` virtual environment is created, we can confirm it's working as expected.
+
+First, completely close out the current shell and open a new one in this app project directory.  This 
```

Comment:
[Non-blocking] I want to make sure I understand how this works. In Step 6, we created a Python virtual environment that uses the x86 Python that we installed _through_ the `x86`/`i386` architecture environment via Rosetta, right?

When we close out of the current x86 shell, how is it that the arm64 terminal can run the Oracle client library?

Reply:
@jonavellecuerdo - excellent question!  That's the beauty and the maddening thing about all this.  All that effort results in a python virtual environment that is compiled for `i386` architecture.  Then, when running `pipenv shell` or `pipenv run <CMD>` and this python installation is used, it's compiled for `i386` architecture.

I dont' want to misspeak here... I haven't quite fully wrapped my head around why you don't need to invoke an `x86` shell to run the `i386` python virtual environment, but in my testing it just doesn't seem to be the case.

I'd like to push on that a bit more for a more thorough understanding.